dashboard/app/helpers/levels_helper.rb

```diff
@@ -28,18 +28,15 @@ def url_from_path(path)
   #   using the value from the `data` param.
   def create_channel(data = {}, src = nil)
 
-    result = ChannelsApi.call(request.env.merge(
-      'REQUEST_METHOD' => 'POST',
-      'PATH_INFO' => '/v3/channels',
-      'REQUEST_PATH' => '/v3/channels',
-      'QUERY_STRING' => src ? "src=#{src}" : '',
-      'CONTENT_TYPE' => 'application/json;charset=utf-8',
-      'rack.input' => StringIO.new(data.to_json)
-    ))
-    headers = result[1]
-
-    # Return the newly created channel ID.
-    headers['Location'].split('/').last
+    storage_app = StorageApps.new(storage_id('user'))
```

Comment:
Awesome! Do we have any indication of whether calling `storage_id('user')` in this context would properly set the storage_id cookie on the rails response? If not, it might be worth adding a safeguard so that no one later adds calls to `create_channel` from a signed-out context, which might cause us to silently lose the cookie. I would propose protecting this method with `return unless current_user` at the beginning and then calling `storage_id_for_user` directly instead of `storage_id`.